src/widgets/tables/merging_sets_screen.py

```diff
@@ -0,0 +1,409 @@
+        document_name = self.database_box.currentText()
+        try:
+            reader = self.loader.create_database_reader(document_name)
+        except ValueError as e:
+            self.import_state_label.setText(str(e))
+        else:
+            self._on_success(reader)
+
+    def _on_success(self, reader, file_path=None):
+        if file_path is not None:
+            self.filepath_line.setText(file_path)
+        self.import_state_label.clear()
+        if self.engine.is_data_big():
+            error = QMessageBox()
+            error.setIcon(QMessageBox.Warning)
+            error.setText("This file is too big.\nYou must save it in database!")
+            error.setWindowTitle("Warning")
+            error.exec_()
+        self.new_data = reader.read(None)
+        last_preview = self.layout.takeAt(2).widget()
+        if last_preview is not None:
+            last_preview.deleteLater()
+        self._render_table(self.new_data_view, False)
+
+        for i in reversed(range(self.columns_left_layout.count() - 1)):
+            # workaround to delete as low content as possible, removing nested layouts was giving me some segfault
+            self.columns_left_layout.itemAt(i).widget().setParent(None)
+
+        for i in reversed(range(self.columns_right_layout.count() - 1)):
+            self.columns_right_layout.itemAt(i).widget().setParent(None)
+
+        imported_columns = self._get_imported_columns()
+        for column in imported_columns:
+            self.columns_left_layout.insertWidget(
+                self.columns_left_layout.count() - 1,
+                ColumnWidget(column, self.columns_left_layout),
+            )
+
+        if self.new_data is not None and self.new_data.columns is not None:
+            for column in sorted(self.new_data.columns, key=lambda x: x.upper()):
+                self.columns_right_layout.insertWidget(
+                    self.columns_right_layout.count() - 1,
+                    ColumnWidget(column, self.columns_right_layout),
+                )
+
+        self.submit_button.setEnabled(self.new_data is not None)
+
+    def _on_submit(self):
+        new_columns_left = []
+        new_columns_right = []
+        for i in range(self.columns_left_layout.count() - 1):
+            new_columns_left.append(
+                self.columns_left_layout.itemAt(i).widget().column_name
+            )
+        for i in range(self.columns_right_layout.count() - 1):
+            new_columns_right.append(
+                self.columns_right_layout.itemAt(i).widget().column_name
+            )
+
+        if len(new_columns_left) != len(new_columns_right):
+            self._render_equality_warning()
+            if self.cancel_merge:
+                self.cancel_merge = False
+                return
+
+        num_columns = max(len(new_columns_left), len(new_columns_right))
+        labels_left = new_columns_left[:num_columns]
+        labels_right = new_columns_right[:num_columns]
```

Comment:
When I tried merge two dataset with iris (in the second I remove 3 columns) I got error. You should add something like this in this place:
`self.new_data = self.new_data[labels_right]`